test/application_system_test_case.rb

```diff
@@ -89,41 +89,70 @@ def teardown
     key = ENV["TEST_DEVICE"].to_sym
     if @@test_devices.key?(key)
       puts "Running tests with the `#{ENV["TEST_DEVICE"]}` device profile specifically.".green
-      @@test_devices = {key => @@test_devices[key]}
+      @@test_devices = @@test_devices.slice(key)
     else
       puts "⚠️ `#{ENV["TEST_DEVICE"]}` isn't a valid device profile in `test/test_helper.rb`, so we'll just run *all* device profiles.".yellow
     end
   end
 
-  def resize_for(display_details)
+  def device_name
+    @device_name ||= @@test_devices.each_key.find { name.include?(_1.to_s) } || raise("unknown test device, you probably want to use `device_test` to generate this test")
+  end
+
+  def display_details
+    @@test_devices[device_name]
+  end
+
+  # Generate a device specific test for each device in `@@test_devices`.
+  #
+  # Automatically resizes the display for the test too.
+  #
+  #   device_test "system test" do
+  #     p device_name
+  #     p display_details
+  #   end
+  def self.device_test(name, &block)
+    @@test_devices.each_key do |device_name|
+      test "#{name} on a #{device_name}" do
+        resize_display # TODO: Figure out why the browser window opens if this is done in `setup`.
```

Comment:
It would be nice if `resize_display` could just become part of `setup` but then the browser window opens, and stays open, for some reason.

Reply:
Yeah, I had this same thought recently! Didn't realize it would cause an issue though...